File: raspberry/SmartHome/app/src/main/java/raspberry/smarthome/database/store/LinkedAccountsStorage.java
package raspberry.smarthome.database.store;

import com.google.firebase.database.DatabaseReference;

import raspberry.smarthome.database.model.LinkedAccounts;
import raspberry.smarthome.database.store.listeners.LinkedAccountsListener;

public interface LinkedAccountsStorage {

    void postLinkedAccounts(LinkedAccounts linkedAccounts);

    void postLinkedAccounts(LinkedAccounts linkedAccounts, DatabaseReference.CompletionListener listener);

    void getLinkedAccounts(LinkedAccountsListener listener);

}

File: smarthomelibary/DataLibrary/datalibrary/src/main/java/smarthome/datalibrary/database/store/LinkedAccountsStorage.kt
package smarthome.datalibrary.database.store

import com.google.android.gms.tasks.OnFailureListener
import com.google.android.gms.tasks.OnSuccessListener
import com.google.firebase.database.DatabaseReference
import smarthome.datalibrary.database.constants.Constants.defFailureListener
import smarthome.datalibrary.database.constants.Constants.defSuccessListener

import smarthome.datalibrary.database.model.LinkedAccounts
import smarthome.datalibrary.database.store.listeners.LinkedAccountsListener

interface LinkedAccountsStorage {
    fun postLinkedAccounts(linkedAccounts: LinkedAccounts, successListener: OnSuccessListener<Void> = defSuccessListener, failureListener: OnFailureListener = defFailureListener)
    fun getLinkedAccounts(listener: LinkedAccountsListener)
}
